packages/lab/src/list-next/useList.ts

```diff
@@ -0,0 +1,250 @@
+    if (lastItem) {
+      focusAndMoveActive(lastItem);
+      updateScroll(lastItem);
+    }
+  };
+
+  const findNextOption = (
+    currentOption: HTMLElement | null,
+    moves: number
+  ): HTMLElement => {
+    const activeOptions = getOptions();
+    // Returns next item, if no current option it will return 0
+    const nextOptionIndex = currentOption
+      ? activeOptions.indexOf(currentOption) + moves
+      : 0;
+    return (
+      activeOptions[nextOptionIndex] || activeOptions[activeOptions.length - 1]
+    );
+  };
+
+  const findPreviousOption = (
+    currentOption: HTMLElement,
+    moves: number
+  ): HTMLElement => {
+    // Return the previous option if it exists; otherwise, returns first option
+    const activeOptions = getOptions();
+    const currentOptionIndex = activeOptions.findIndex(
+      (i) => i.id === currentOption.id
+    );
+    return activeOptions[currentOptionIndex - moves] || activeOptions[0];
+  };
+
+  // CONTEXT CALLBACKS
+  const select = useCallback(
+    (event: SyntheticEvent<HTMLLIElement>) => {
+      const newValue = event.currentTarget.dataset.value;
+      if (event.type === "click") {
+        setShowFocusRing(false);
+      }
+      const activeOptions = getOptions();
+      const isActiveOption =
+        activeOptions.findIndex((i) => i.id === event.currentTarget.id) !== -1;
+      if (newValue && selectedItem !== newValue && isActiveOption) {
+        selectItem(event.currentTarget);
+        onChange?.(event, { value: newValue });
+      }
+    },
+    [selectItem, selectedItem, onChange, getOptions]
+  );
+
+  const isSelected = useCallback(
+    (id: string) => selectedItem === id,
+    [selectedItem]
+  );
+
+  const isFocused = useCallback(
+    (id: string | undefined) => activeDescendant === id && showFocusRing,
+    [activeDescendant, showFocusRing]
+  );
+
+  const getActiveItem = () => {
+    const activeOptions = getOptions();
+    const activeIndex = activeOptions.findIndex(
+      (i) => i.id === activeDescendant
+    );
+    return activeOptions[activeIndex !== -1 ? activeIndex : 0];
+  };
+
+  // HANDLERS
+  const handleBlur = () => {
+    setShowFocusRing(false);
+  };
+
+  const handleMouseDown = () => {
+    // When list gets focused, we can't guarantee that focus happens after click event.
+    // If first focus (where !activeDescendant) happens from a click, list shouldn't render focus ring in the first element.
+    setShowFocusRing(false);
+  };
+
+  // takes care of focus when using keyboard navigation
+  const handleFocus = () => {
+    const activeElement = getActiveItem();
+    focusAndMoveActive(activeElement);
+  };
+
+  // takes care of keydown when using keyboard navigation
+  const handleKeyDown = (event: KeyboardEvent<HTMLUListElement>) => {
+    const { key } = event;
+    const currentItem = getActiveItem();
+    let nextItem = currentItem;
+    if (!currentItem) {
+      return;
+    }
+    setShowFocusRing(true);
```

Comment:
shouldn't this happen only when focus happened?

Reply:
nice catch! yeah, then i can default to false, rely on focus and forget about this ⭐ ta